sources/standard/inbox/__init__.py

```diff
@@ -0,0 +1,190 @@
+            if len(filter_emails) == 1:
+                filter_emails = filter_emails[0]
+            if isinstance(filter_emails, str):
+                criteria.append(f"(FROM {filter_emails})")
+            else:
+                email_filter = " ".join([f"FROM {email}" for email in filter_emails])
+                criteria.append(f"(OR {email_filter})")
+
+        uids = get_message_uids(client, criteria)
+        for i in range(0, len(uids), chucksize):
+            yield uids[i : i + chucksize]
+
+
+class ImapSource(FilesystemSource):
+    pass
+
+
+@dlt.transformer(
+    name="attachments",
+    write_disposition="merge",
+    merge_key="data_hash",
+    primary_key="data_hash",
+)
+def get_attachments_by_uid(
+    items: TDataItems,
+    storage_path: str,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    filter_by_mime_type: Sequence[str] = (),
+) -> TDataItem:
+    """Downloads attachments from email messages based on the provided message UIDs.
+
+    Args:
+        items (TDataItems): An iterable containing dictionaries with 'message_uid' representing the email message UIDs.
+        storage_path (str): The local folder path where attachments will be downloaded.
+        host (str, optional): The hostname of the IMAP server. Default is 'dlt.secrets.value'.
+        email_account (str, optional): The email account used to log in to the IMAP server. Default is 'dlt.secrets.value'.
+        password (str, optional): The password for the email account. Default is 'dlt.secrets.value'.
+        filter_by_mime_type (Sequence[str], optional): A sequence of MIME types used to filter attachments based on their content type. Default is an empty sequence.
+
+    Yields:
+        TDataItem: A dictionary containing the collected email information and attachment details.
+    """
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select()
+
+        for item in items:
+            message_uid = str(item["message_uid"])
+            msg = get_message(client, message_uid)
+            if isinstance(filter_by_mime_type, str):
+                filter_by_mime_type = [filter_by_mime_type]
+            if filter_by_mime_type:
+                attachments = None
+                for mime_type in filter_by_mime_type:
+                    new_attachments = extract_attachments(msg, mime_type)
+                    if not attachments:
+                        attachments = new_attachments
+                    else:
+                        attachments = chain(attachments, new_attachments)
+            email_info = extract_email_info(msg)
+
+            for attachment in attachments:
+                file_data = ImapSource(
+                    file_name=attachment["file_name"],
+                    storage_path=storage_path,
+                    file=attachment["payload"],
+                    mod_date=email_info["Date"],
```

Comment:
we should never use `Date`. it may be fake, use internal timestamp as before.

Reply:
I am moving it back to the INTERNALDATE, I used it just to avoid doing an other request to the server just to get the date, but it looks like it is really a [different](https://www.metaspike.com/imap-internal-date-forensic-email-authentication/#:~:text=Internal%20Date%20is%20an%20IMAP,Let's%20start%20with%20an%20example.)